tools/eventgen/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import(
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/BurntSushi/toml"
+)
+
+// Function for generating the front matter of an event file
+// and the details of the event before the description.
+func generateFrontMatter(event Event) string {
+	// Create the internal URL slug.
+	url := fmt.Sprintf("/events/%s", event.URLSlug)
+
+	// Parse the start and end dates, then format them.
+	startDate, _ := time.Parse("2006-01-02", event.StartDate)
+	endDate, _ := time.Parse("2006-01-02", event.EndDate)
+	formattedStartDate := startDate.Format("January 02")
+	formattedEndDate := endDate.Format("January 02")
+
+	// Grab the year of both the start and end dates to determine
+	// if the dates are in different years.
+	startYear := startDate.Format("2006")
+	endYear := endDate.Format("2006")
+
+	// Assign the date variable. If it is a one day event just show the start date. If the start
+	// date is not equal to the end date, display both years with the dates. Otherwise display
+	// the start and end date with the end year at the end of the string.
+	var date string
+	if formattedStartDate == formattedEndDate {
+		date = fmt.Sprintf("%s, %s", formattedStartDate, startYear)
+	} else if startYear != endYear {
+		date = fmt.Sprintf("%s, %s - %s, %s", formattedStartDate, startYear, formattedEndDate, endYear)
+	} else {
+		date = fmt.Sprintf("%s - %s, %s", formattedStartDate, formattedEndDate, endYear)
+	}
+
+	frontMatter := fmt.Sprintf(`---
```

Comment:
This is fine for now, but in the future, this format string is perhaps complicated enough that you'd want to just break down and use Go's built-in text templating stack. As it would make the rendering a little less error prone. (No need to remember the ordering of the `%s` and instead just use `{{ title }}` and `{{ url }}`, etc.)
https://golang.org/pkg/text/template/